drivers/serial/uart_nrfx_uarte.c

```diff
@@ -1534,6 +1539,8 @@ static void uarte_nrfx_set_power_state(const struct device *dev,
 			NRF_UARTE_PSEL_DISCONNECTED) {
 			nrf_uarte_task_trigger(uarte, NRF_UARTE_TASK_STARTRX);
 		}
+
+		data->pm_state = new_state;
```

Comment:
Shouldn't this be inside an ifdef? I think that data->pm_state is only available when CONFIG_DEVICE_POWER_MANAGEMENT is set

Reply:
whole function is conditionally compiled

Reply:
ahh. sorry, folding again. it showed like it is a part of uarte_nrfx_poll_out function.